### PR TITLE
Save `planned:false` for ad-hoc LOIs.

### LINKED ISSUE #2300: [Add LOI] Add a flag to ad hoc LOIs to differentiate from predefined ones
To add the following to imported LOIs:
```
planned: false
```

See also https://github.com/google/ground-platform/issues/1556

### PR DESCRIPTION
**Pull request by sufyanAbbasi**
Fixes #2300.

Send `planned: false` for LOIs that are created ad-hoc from the app. Maps to `isPlanned` on the client.

- [x] Adds the `planned` field to the `LoiDocument` for Firsetore storage, sets to false when an ad-hoc request is made.
- [x] Maps the field to `isPlanned` in the `LocationOfInterest` and `LocationOfInterestMutation`
- [x] Refactored usage of `isOpportunistic` and to `isPlanned`.
- [x] Added support for the local datastore schema.
- [x] Unit tests.

- Breakpoints at document load time verified that the Firestore was storing "planned:false" for newly created entries and reserialized back on the client.
- Firestore stores the field (see image).

Screenshot 2024-03-06 at 1 24 17 AM

@gino-m PTAL!

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt [hunk 1]
```diff
@@ -55,6 +55,8 @@ data class LocationOfInterest(
   val isOpportunistic: Boolean = false,
   /** Custom map of properties for this LOI. Corresponds to the properties field in GeoJSON */
   val properties: LoiProperties = mapOf(),
+  /** Whether the LOI was predefined in the survey or not (i.e. added ad hoc). */
+  val planned: Boolean? = null,
```
Comment (anandwana001) [inline ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt:59]: how about naming it as `isPlanned`? 
WDYT?

**Review by gino-m — CHANGES_REQUESTED** [review]
Hey @sufyanAbbasi ! I apologize; I didn't realize we already had `isOpportunistic` internally because nothing is getting written to the remote db. Can we rename `isOpportunistic` to `isPlanned`, invert the condition, and use that to populate the similarly named value in `LoIDocument`? Sorry for the churn.

**Comment by sufyanAbbasi** [comment]
Sounds like a plan. Just to be clear, do you want the Firebase store to have `isPlanned` or `planned`?

**Review by gino-m — COMMENTED** [review]
👏 ! Great, thank you!

File: ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt (hunk 1 above)
Comment (sufyanAbbasi) [reply ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt]: Thank you! Done.

**Review by gino-m — APPROVED** [review]

File: ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/LoiMutationConverterTest.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
+package com.google.android.ground.persistence.remote.firebase.schema
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/LoiMutationConverterTest.kt:1]: Needs a license header

**Comment by gino-m** [comment]
detekt is failing, run `gradle checkCode` locally see output of detekt and other check code tasks.

File: ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/LoiMutationConverterTest.kt (hunk 2 above)
Comment (sufyanAbbasi) [reply ground/src/test/java/com/google/android/ground/persistence/remote/firebase/schema/LoiMutationConverterTest.kt:17]: Done.

**Comment by sufyanAbbasi** [comment]
Thank you! Fixed now.

**Pull request merged**